src/runtime/pybuffer.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Python.Runtime
+{
+    public class PyBuffer
+    {
+        private IntPtr _handle;
+        private Runtime.Py_buffer _view;
+        private GCHandle _gchandle;
+
+        internal PyBuffer(IntPtr obj, int flags, out bool success)
+        {
+            int size = Marshal.SizeOf(typeof(Runtime.Py_buffer));
+            byte[] rawData = new byte[size];
+            _gchandle = GCHandle.Alloc(rawData, GCHandleType.Pinned);
+            _handle = _gchandle.AddrOfPinnedObject();
+            _view = (Runtime.Py_buffer)Marshal.PtrToStructure(_handle, typeof(Runtime.Py_buffer));
```

Comment:
This is the only line where `_view` is initialized, however, at this moment the memory `_handle` points to is empty (e.g. all zeroes). So the `_view` is also always empty.
None of the `if`s below ever succeed.

Reply:
Your wrong. `_view` _should_ be empty, because PyObject_GetBuffer fills `_view` with data. Every `if` below works as expected

Reply:
@SnGmng you might be using some funky runtime, which makes this happen. `PyObject_GetBuffer` does not get a pointer to `_view`. It gets a pointer to the `rawData` instead. The call to `Marshal.PtrToStructure` simply creates a copy of memory it is pointing to.

Reply:
Oh my bad i messed up the order of these two lines when i was moving the code into the constructor. The PR was working before this commit tho.